pkg/aws/eni/types/types.go

```diff
@@ -66,33 +66,33 @@ type ENISpec struct {
 	//
 	// OBSOLETE: This field is obsolete, please use Spec.IPAM.MaxAboveWatermark
 	//
-	// +optional
+	// +kubebuilder:validation:Minimum=0
 	MaxAboveWatermark int `json:"max-above-watermark,omitempty"`
 
 	// FirstInterfaceIndex is the index of the first ENI to use for IP
 	// allocation, e.g. if the node has eth0, eth1, eth2 and
 	// FirstInterfaceIndex is set to 1, then only eth1 and eth2 will be
 	// used for IP allocation, eth0 will be ignored for PodIP allocation.
 	//
-	// +optional
+	// +kubebuilder:validation:Minimum=0
```

Comment:
This value (and the ones that you added the markers) are optional, will the validation complain if the values are not set? (not set  != "0")

Reply:
as the type is *int, should it be just optional ? Example from kubebuilder repo https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/cronjob-tutorial/testdata/project/api/v1/cronjob_types.go#L92

Reply:
It shouldn't complain because this validation is not new, see here: https://github.com/cilium/cilium/pull/11476/commits/90cc78f194799a42a4708c01932820de04dad60b#diff-da7eac65a6ed1ef87f8fec512cb90529R330-R333

Reply:
I can mark the types as explicitly optional so that it is clear.